src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,281 @@
+        updated_at: the datetime when the workspace was last updated.
 
     Examples:
         >>> from argilla import rg
-        >>> workspace = rg.Workspace.from_name("my-workspace")
-        >>> workspace.id
+        >>> workspace = rg.Workspace.from_name("my-workspace") # or `Workspace.from_id("...")`
+        >>> workspace.add_user("my-user")
+        >>> print(workspace.users)
+        [WorkspaceUserModel(id='...', first_name='Luke', last_name="Skywalker', full_name='Luke Skywalker', username='my-user', role='annotator', workspaces=['my-workspace'], api_key='...', inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0), updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0))]
+        >>> workspace.delete_user("my-user")
+        >>> print(workspace.users)
+        []
     """
 
-    def __init__(self, name: str, *, id: Optional[str] = None) -> None:
-        """Initializes a new `Workspace` instance using the provided name. If no name
-        is provided, the current workspace will be used.
+    client: "httpx.Client"
+    id: UUID
+    name: str
+    users: Optional[List["WorkspaceUserModel"]] = None
+    inserted_at: datetime
+    updated_at: datetime
+
+    def __init__(
+        self, client: "httpx.Client", *, id: Union[str, UUID], name: str, inserted_at: datetime, updated_at: datetime
+    ) -> None:
+        """Initializes a `Workspace` instance. Ideally this class should not be instantiated
+        directly, instead you should use one of the following classmethods:
+            - `Workspace.create` to create a new workspace in Argilla
+            - `Workspace.from_name` to get an existing workspace from Argilla by its name
+            - `Workspace.from_id` to get an existing workspace from Argilla by its ID
 
         Args:
-            name: the name of the workspace to be managed. If not provided, the current
-                workspace will be used.
+            client: the `httpx.Client` initialized to interact with the Argilla API.
+            name: the name of the workspace to be managed.
             id: the ID of the workspace to be managed.
+            inserted_at: the datetime when the workspace was created.
+            updated_at: the datetime when the workspace was last updated.
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace(
+            ...     client=rg.active_client().http_client.httpx,
+            ...     name="my-workspace",
+            ...     id="...",
+            ...     inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ...     updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ... )
         """
+        self.client = client
```

Comment:
I think we can keep this attribute private, since won't be used outside this class
```suggestion
        self._client = client
```

Reply:
Ok I'll make it private then! 👍🏻